### ISSUE TITLE
Feature Request: Presets for Splitting

### ISSUE THREAD

**Issue by mirobo** [stemrollerapp/stemroller#72]
I use StemRoller for practicing songs so I usually want a track with bass and the track without bass (vice versa for drums).

Right now I open the stems in Reaper and export new renders from there. I would be cool if this could be done directly from within StemRoller :-)

**Comment by pan-danio** [comment]
+1 
Drums is the only thing i need in my use case. It will be nice to extract just drums from selected files (or any track selected from available). This will hopefully speed up extracting when proessing multiple files

**Comment by mirobo** [comment]
> +1 Drums is the only thing i need in my use case. It will be nice to extract just drums from selected files (or any track selected from available). This will hopefully speed up extracting when proessing multiple files

Good idea with track selection. Maybe "presets for splitting" is a bad idea but just having checkboxes on der main UI where you can select what you want to extract from it may be enough.

Maybe something like this?

I added "All Tracks, but without.." twice, so you could export more than just one "configuration" from it.
Example: 
- single track of drums
- single track of Bass
- full song without bass
- full song without drums

![image](https://github.com/user-attachments/assets/62c087ec-f4cf-4be5-8725-ac1ed0e1fcb7)

**Comment by iffyloop** [comment]
I'd currently consider this outside the scope of StemRoller, but if I have more time in the future I might work on this. Will leave the issue open though in case others are interested and want to comment.

**Comment by iffyloop** [comment]
Changing the title - let's use this as a mega-issue for other DAW-like functionality. I would like to add custom mixes and timestretching together in a (potential) complete redesign of StemRoller. I think the current code is not well-adapted to custom audio processing - relies too much on external commands to ffmpeg (which works for the original use case described by @mirobo, but if we're supporting custom mixes it'd probably be ideal also to support a real-time previewer with adjustable volume for each stem).

**Comment by DrFangaz** [comment]
Also, adding a feature to stem the drums and vocals parts into separate splits would be awesome. Ex. (Drums - kick, snare, hihat, crash. Vocals - Lead, background harmonies.)
 YouTube playlist saving and restore due to download error.
